parser-sdk/nodejs/parser-utils.js

```diff
@@ -0,0 +1,60 @@
+const fs = require("fs");
+const Ajv = require("ajv-draft-04");
+const ajv = new Ajv();
+const addFormats = require("ajv-formats");
+addFormats(ajv);
+const util = require("util");
+var jsonpointer = require("jsonpointer");
+
+// eslint-disable-next-line security/detect-non-literal-fs-filename
+const readFile = util.promisify(fs.readFile);
+const { v4: uuid } = require("uuid");
+
+function addIdsAndDates(findings) {
+  return findings.map((finding) => {
+    return {
+      ...finding,
+      id: uuid(),
+      parsed_at: new Date().toISOString(),
+    };
+  });
+}
+
+// eslint-disable-next-line
+async function validateAgainstJsonSchema(jsonData) {
+  const jsonSchemaString = await readFile(
+    __dirname + "/findings-schema.json",
+    "utf8"
+  );
+  const jsonSchema = JSON.parse(jsonSchemaString);
+  const validator = ajv.compile(jsonSchema);
+  const valid = validator(jsonData);
+  if (!valid) {
+    const errorMessage = generateErrorMessage(validator.errors, jsonData);
+    throw new Error(errorMessage);
+  }
+}
+
+// eslint-disable-next-line
+async function addSampleIdsAndDatesAndValidate(jsonData) {
+  // add sample IDs and Dates only if the jsonData Array is not empty
+  if (jsonData.length > 0) {
```

Comment:
Is this required?
Should the `addIdsAndDates` function also work fine with arrays with zero entries?

Reply:
it is not needed anymore and I removed it.